File: js/bubble_sort.js
// input dependencies
const prompt = require('readline-sync');
const fs = require('fs');

let direc = process.argv.slice(2)[0];
let txt = fs.readFileSync(`./data/${direc}.txt`).toString();

if (txt.length === 0) {
	txt = prompt.question();
}


let array = txt.split(',').map((a) => {
	return parseInt(a.trim(), 10);
});

function main() {
	let len_arr = array.length;

	_start = process.hrtime();
	// algo start
	for (let i = len_arr; i >= 0; i--) {
		for (let il = 0; il < i - 1; il++) {
			if (array[il] > array[il+1]) {
				[ array[il], array[il+1] ] = [ array[il+1], array[il] ]
			}
		}
	}
	_end = process.hrtime();
}

main();

fs.writeFileSync("./result/js.txt", `bubble_sort, ${direc}, ${(_end[0] + _end[1]/1000000000) - (_start[0] + _start[1]/1000000000)}\n`, {encoding: 'utf8', flag:'a'});

File: py/bubble_sort.py
import math
import sys
import time


if __name__ == "__main__":
	
	direc = sys.argv[1]

	with open(f"./data/{direc}.txt", "r") as f:
		txt = f.read()
	
	if not txt:
		txt = input()

	array = [int(x.strip()) for x in txt.split(',')]

	array_n = len(array)

	_start = time.time()
	# algo start
	found = False
	for i in range(array_n-1, -1, -1):
		for il in range(0, i):
			if array[il] > array[il+1]:
				array[il], array[il+1] = array[il+1], array[il]
				found = True
			if not found: break
	_end = time.time()

	with open(f"./result/py.txt", "a") as f:
		f.write(f"bubble_sort, {direc}, {_end - _start}\n")